File: 이코테/P115_왕실의나이트/P115.py
move = [ [2,1],[-2,1],[2,-1],[-2,-1], [1,2],[-1,2],[1,-2],[-1,-2]]

y, x = list(input())
x = int(x)

result = 0

for i in range(8):
    
    if( 1 <= x + move[i][1] and x + move[i][1] <= 8 and ord('a') < ord(y) + move[i][0] and ord(y) + move[i][0] <= ord('h') ):
        result+=1
        
print(result)

File: 이코테/P115_왕실의나이트/P115.c
#include<stdio.h>
int main()
{
    char x;
    int y;
    int move[8][2] = { {2,1},{-2,1},{2,-1},{-2,-1},{1,2},{-1,2},{1,-2},{-1,-2} };
    scanf("%c%d",&x,&y);


    int result = 0;
    for(int i=0; i < 8 ; i++)
    {
        if( 'a'<=x+move[i][1] && x+move[i][1]<='h' && 1 <= y+move[i][0] && y+move[i][0] <= 8   ) result++;
        
    }

    printf("%d",result);

}